PostHog/Utils/UIApplication+.swift

```diff
@@ -0,0 +1,31 @@
+//
+//  UIApplication+.swift
+//  PostHog
+//
+//  Created by Yiannis Josephides on 11/11/2024.
+//
+
+import UIKit
+
+extension UIApplication {
+    static func getCurrentWindow(checkForegrounded: Bool = false) -> UIWindow? {
```

Comment:
I think we could invert the default value here, the normal is to check the current window, which means active by default (and in the foreground).
Only events captured in the background thread would have to skip this check because there's no active window but we still want its metadata, but that hs the exception.